src/plugins/mer/merbuildconfiguration.cpp

```diff
@@ -73,6 +78,19 @@ MerBuildConfiguration::MerBuildConfiguration(Target *target, Core::Id id)
     QmakeProject *qmakeProject = static_cast<QmakeProject *>(project());
     connect(qmakeProject, &QmakeProject::buildDirectoryInitialized,
             this, &MerBuildConfiguration::setupExtraParserArguments);
+
+    connect(SessionManager::instance(), &SessionManager::projectAdded,
+            this, &MerBuildConfiguration::setupExtraParserArguments);
+
+    connect(EditorManager::instance(), &EditorManager::aboutToSave,
+            project(), [this](IDocument *document) {
```

Comment:
Using `this` as the `context` here would be safer I think. To me it seems likely that `project()` will live longer than `this`.

Reply:
Removed: now implemented `fromMap` and `toMap`, so this connection is not needed.